src/app/app-routing.module.ts

```diff
@@ -0,0 +1,31 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { DetailsFormComponent } from './screen/onboarding/details-form/details-form.component';
+import { HomeComponent } from './components/home/home.component';
+import { RestaurantAddedPopupComponent } from './screen/onboarding/restaurant-added-popup/restaurant-added-popup.component';
+import { AllRestaurantsComponent } from './screen/onboarding/all-restaurants/all-restaurants.component';
+
+const routes: Routes = [
+  {
+    path: 'onboarding/details-form',
+    component: DetailsFormComponent
+  },
+  {
+    path: 'home',
+    component: HomeComponent
+  },
+  {
+    path: 'popup',
+    component: RestaurantAddedPopupComponent
```

Comment:
every component does not need a route. we need routes only where we need to redirect to some screen

Reply:
to show on button click , i have created new component , otherwise i have to use conditional rendering of popup component